src/rest/connector/libs/nd/implementation.py

```diff
@@ -0,0 +1,310 @@
+import json
+import logging
+import requests
+import time
+import urllib.parse
+from inspect import signature
+
+from requests.exceptions import RequestException
+
+from pyats.connections import BaseConnection
+from rest.connector.implementation import Implementation as Imp
+from rest.connector.utils import get_username_password
+
+# create a logger for this module
+log = logging.getLogger(__name__)
+
+
+class Implementation(Imp):
+    """Rest Implementation for ND
+    Implementation of Rest connection to devices based on pyATS BaseConnection
+    for ND
+
+    YAML Example
+    ------------
+
+        devices:
+            nd1:
+                platform: nd
+                connections:
+                    rest:
+                        class: rest.connector.Rest
+                        ip : "2.3.4.5"
+                credentials:
+                    rest:
+                        username: admin
+                        password: cisco123
+
+    Code Example
+    ------------
+        >>> from pyats.topology import loader
+        >>> testbed = loader.load('/users/xxx/xxx/testbed.yaml')
+        >>> device = testbed.devices['nd1']
+        >>> device.connect(alias='rest', via='rest')
+        >>> device.rest.connected
+        True
+    """
+
+    @BaseConnection.locked
+    def connect(self, timeout=30, retries=3, retry_wait=10):
+        """connect to the device via REST
+        Arguments
+        ---------
+            timeout (int): Timeout value
+            retries (int): Max retries on request exception (default: 3)
+            retry_wait (int): Seconds to wait before retry (default: 10)
+
+        Raises
+        ------
+
+        Exception
+        ---------
+            If the connection did not go well
+
+        Note
+        ----
+        There is no return from this method. If something goes wrong, an
+        exception will be raised.
+
+        YAML Example
+        ------------
+            devices:
+                nd1:
+                    platform: nd
```

Comment:
same

Reply:
explained in a reply to the initial comment - platform is the correct abstraction selector for ND

Reply:
same. `os: nd` is better.

Reply:
replied on other comment thread.

Reply:
Done